### PR TITLE
Render Enum params correctly in OpenAPI

### PR DESCRIPTION
**Pull request by dialvarezs**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [ ] Have you got 100% test coverage on new code?
- [ ] Have you updated the prose documentation?
- [ ] Have you updated the reference documentation?

Implements #975.

### CODE REVIEW

File: starlite/openapi/schema.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ class GenericPydanticSchema(OpenAPI310PydanticSchema):
 
 def get_schema_for_field_type(field: ModelField, plugins: List["PluginProtocol"]) -> Schema:
     """Get or create a Schema object for the given field type."""
-    field_type = field.outer_type_
+    field_type = field.type_ if issubclass(field.type_, Enum) else field.outer_type_
```
Comment (Goldziher) [inline starlite/openapi/schema.py:201]: use the helper `is_class_and_subclass`

**Review by Goldziher — COMMENTED** [review]
Can you add a test to ensure the issue is resokved?

**Comment by dialvarezs** [comment]
Done!

**Comment by Goldziher** [comment]
@all-contributors add @dialvarezs code

**Pull request merged**